torch/csrc/jit/script/compiler.cpp

```diff
@@ -641,20 +641,18 @@ c10::optional<MatchedSchema> tryMatchSchema(
     if (!positional)
       return c10::nullopt;
     positional_inputs.push_back(positional);
-
-    if (schema.is_vararg() && schema_i == schema.arguments().size() - 1) {
-      // Freeze iteration to the last argument in the schema and keep going for
-      // all of the provided arguments
-      if (used_args < modifiedArgs.size()) {
-        schema_i--;
-      }
-    }
   }
   // check for unused self argument
   if(self != c10::nullopt) {
     err() << "provided self argument not used in schema\n";
   }
 
+  if (schema.is_vararg()) {
+    for(;used_args < modifiedArgs.size(); ++used_args) {
+      positional_inputs.push_back(modifiedArgs[used_args].value(graph));
```

Comment:
Looks like this skips tryMatchArgument logic for the left args, curious if this will have some issues (e.g. if the remaining arg be a convertible type or Nonetype, we might need to emit special node instead).

Reply:
yes, this is done intentionally because we expect that varargs functions can accept any input value. This is true of all of our varargs functions today (prim ops, print, and format), and I'd like to avoid adding new vararg functions in the future if possible. Note that any problem that would occur with `format` would also already be a problem in `print` so if there are missing options for printing a None type, then we should add them.